### DISCUSSION TITLE
Toolpad in HTML branding and navigation not working

### DISCUSSION THREAD

**Discussion by sjwen98** [mui/toolpad#4440] [Help and support]
Toolpad in HTML branding and navigation not working

I am trying to use toolpad without using node but in pure html and javascript.
The branding in AppProvider didn't work but npm project yes.

```html
<head>
    <script type="importmap">
        {
            "imports": {
                "react": "//esm.sh/react@18.3.1",
                "react-dom": "//esm.sh/react-dom@18.3.1",
                "@mui/material": "//esm.sh/@mui/material@6.1.7",
                "@mui/lab/Masonry": "//esm.sh/@mui/lab@6.0.0-beta.15/Masonry",
                "@mui/icons-material": "//esm.sh/@mui/icons-material@6.1.7",
                "@emotion/react": "//esm.sh/@emotion/react@11.13.3",
                "@emotion/styled": "//esm.sh/@emotion/styled@11.13.0",
                "@toolpad/core": "//esm.sh/@toolpad/core@0.9.0"
            }
        }
    </script>
</head>

<div id="root"></div>
<script src="//unpkg.com/@babel/standalone/babel.min.js" ></script>
<script type="text/babel" data-type="module">
    import * as React from "react";
    import { createRoot } from "react-dom";
    import { Box } from "@mui/material";
    import { AppProvider, DashboardLayout, PageContainer } from "@toolpad/core"

    function App() {
        return (
            <AppProvider branding={{ logo: <img src="https://mui.com/static/logo.png" alt="MUI logo" />, title: 'MUI' }}>
                <DashboardLayout>
                    <PageContainer>
                        <Box>Hello World!</Box>
                    </PageContainer>
                </DashboardLayout>
            </AppProvider>
        );
    }

    const root = createRoot(document.getElementById("root"));
    root.render(<App />);
</script>`
```
and here is the output
![Screenshot from 2024-11-19 17-02-50](https://github.com/user-attachments/assets/eb64d245-1ea9-4e06-a310-ccd393f8a797)

expected

![Screenshot from 2024-11-19 17-04-26](https://github.com/user-attachments/assets/d38b2525-276a-4835-8124-12d9dbab2698)

here is the [CodeSandbox](https://codesandbox.io/p/sandbox/toolpad-in-html-cd2ytx)

**Comment by bharatkashyap** [comment]
Debugging might be easier if you could create a codesandbox reproduction

**Reply by sjwen98** [reply]
Thanks, added.

**Reply by bharatkashyap** [reply]
Perhaps it is different for you but for me nothing loads: <img width="1153" alt="Screenshot 2024-11-20 at 11 41 46 AM" src="https://github.com/user-attachments/assets/54d6d14d-e1d1-4a2b-a5fd-902238f3f3d9">

**Reply by sjwen98** [reply]
my side got output

![Screenshot from 2024-11-20 14-37-03](https://github.com/user-attachments/assets/5c649550-f3b7-49bc-8bfd-ace55987cef1)

**Reply by apedroferreira** [reply]
I'm not very familiar with this type of setup but it seems that features that use React context under the hood aren't working in it...
I've tried just a simple example with MUI theming and it doesn't seem to work either.

**Reply by oliviertassinari** [reply]
https://github.com/mui/material-ui/tree/master/examples/material-ui-via-cdn

**Reply by sjwen98** [reply]
@oliviertassinari Yes, I am following that example, but the issue is that Toolpad's 'branding,' 'navigation,' and other props aren't working.

**Reply by apedroferreira** [reply]
Does changing colors in the theme work in that example?
I tried to paste it in codesandbox, changing the primary color in the theme to change a `Button`s color, and it doesn't seem to work for me...

**Reply by sjwen98** [reply]
That isn't working either. None of the AppProvider's attributes are working.
